edxval/views.py

```diff
@@ -99,6 +103,52 @@ class SubtitleDetail(MultipleFieldLookupMixin, generics.RetrieveUpdateDestroyAPI
     serializer_class = SubtitleSerializer
 
 
+class VideoImagesView(APIView):
+    """
+    View to update course video images.
+    """
+    authentication_classes = (OAuth2Authentication, SessionAuthentication)
+
+    def post(self, request):
+        """
+        Update a course video image instance with auto generated image names.
+        """
+        attrs = ('course_id', 'generated_images')
+        missing = [attr for attr in attrs if attr not in request.data]
+        if missing:
+            return Response(
+                status=status.HTTP_400_BAD_REQUEST,
+                data={
+                    'message': u'{missing} must be specified to update a video image.'.format(
+                        missing=' and '.join(missing)
+                    )
+                }
+            )
+
+        course_id = request.data['course_id']
+        generated_images = request.data['generated_images']
+
+        try:
+            course_video = CourseVideo.objects.select_related('video_image').get(
+                course_id=unicode(course_id)
+            )
+        except CourseVideo.DoesNotExist:
+            return Response(
+                status=status.HTTP_400_BAD_REQUEST,
+                data={'message': u'CourseVideo not found for course_id: {course_id}'.format(course_id=course_id)}
+            )
+
+        try:
+            VideoImage.create_or_update(course_video, None, generated_images=generated_images)
+        except ValidationError as ex:
+            return Response(
+                status=status.HTTP_400_BAD_REQUEST,
+                data={'message': str(ex)}
+            )
+
+        return Response(generated_images)
```

Comment:
I am not sure what exactly we should return. I have changed this and now only return empty response.